modules/wr_fronius/fronius.py

```diff
@@ -0,0 +1,123 @@
+            pvkwh_start = int(f.read())
+            pvkwh_new = pvkwh_start + pvday + pvkwh_offset
+            if pvkwh_new > pvkwh:
+                if pvkwh_new - pvkwh >= 100:
+                    # Korregiere Abweichung
+                    pvkwh_diff = pvkwh_new - pvkwh - 99
+                    pvkwh_offset -= pvkwh_diff
+                    pvkwh_new -= pvkwh_diff
+                pvkwh = pvkwh_new
+            else:
+                # Berechne Abweichung als Mittelwert von aktueller und bisheriger Abweichung
+                pvkwh_offset = round((pvkwh_offset + pvkwh - pvkwh_start - pvday) / 2)
+    except FileNotFoundError as e:
+        DebugLog(1, str(e))
+
+if wrfronius2ip != "none":
+    try:
+        response = requests.get('http://'+wrfronius2ip+'/solar_api/v1/GetPowerFlowRealtimeData.fcgi', params=params, timeout=3)
+        pv2watttmp = response.json()
+        DebugLog(1, 'response: ' + str(response))
+        DebugLog(2, 'response_pv2watt: ' + str(pv2watttmp))
+        pv2watt = int(pv2watttmp["Body"]["Data"]["Site"]["P_PV"])
+    except:
+        # Wenn WR aus bzw. im Standby (keine Antwort), ersetze leeren Wert durch eine 0
+        pv2watt = 0
+    pvwatt = (pvwatt + pv2watt) * -1
+    DebugLog(2, 'pvwatt: ' + str(pvwatt))
+    # Zur weiteren Verwendung im Webinterface
+    with open("/var/www/html/openWB/ramdisk/pvwatt", "w") as f:
+        f.write(str(pvwatt))
+    if wrfroniusisgen24 == "0":
+        try:
+            pv2kwh = int(pv2watttmp["Body"]["Data"]["Site"]["E_Total"])
+        except:
+            pv2kwh = 0
+        pvgkwh = pvkwh + pv2kwh
+        if pvgkwh > 0:
+            with open("/var/www/html/openWB/ramdisk/pvkwh", "w") as f:
+                f.write(str(pvgkwh))
+            with open("/var/www/html/openWB/ramdisk/pvkwhk", "w") as f:
+                f.write('{:.3f}'.format(round(pvgkwh / 1000, 3)))
+else:
+    pvwatt *= -1
+    DebugLog(2, 'pvwatt: ' + str(pvwatt))
+    # Zur weiteren Verwendung im Webinterface
+    with open("/var/www/html/openWB/ramdisk/pvwatt", "w") as f:
+        f.write(str(pvwatt))
+    if wrfroniusisgen24 == "0" and pvkwh > 0:
+        if pvday == 0 and pvkwh > pvkwh_start + pvkwh_offset:
+            with open("/var/www/html/openWB/ramdisk/pvkwh_start", "w") as f:
+                f.write(str(pvkwh))
+            with open("/var/www/html/openWB/ramdisk/pvkwh_offset", "w") as f:
+                with open("/var/www/html/openWB/ramdisk/pvkwh", "r") as ff:
+                    pvkwh_offset = int(ff.read()) - pvkwh
+                    pvkwh += pvkwh_offset
+                    f.write(str(pvkwh_offset))
+        else:
+            with open("/var/www/html/openWB/ramdisk/pvkwh_offset", "w") as f:
+                f.write(str(pvkwh_offset))
+        with open("/var/www/html/openWB/ramdisk/pvkwh", "w") as f:
+            f.write(str(pvkwh))
+        with open("/var/www/html/openWB/ramdisk/pvkwhk", "w") as f:
+            f.write('{:.3f}'.format(round(pvkwh / 1000, 3)))
```

Comment:
Das Runden ist doppelt. `{:.3f}` rundet UND das `round` rundet. Eins von beiden kann weg.

`'{:.3f}'` lässt trailing zeroes drin, `'{:.3g}'` würde das nicht tun.

(In Zeile 99 nochmal das Gleiche)

Reply:
Ok. Die 3 Nachkommastellen sollten hier garantiert werden.